### ISSUE TITLE
L027 rule failing on lambda with several tables join

### ISSUE THREAD

**Issue by KulykDmytro** [sqlfluff/sqlfluff#3961]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using lambda functions (f.e. `filter`) linter is failing on expression definition
Expecting that identifier should be full qualified

### Expected Behaviour

Expressions in lambda are referenced to inbound array items and not expected to be full qualified

### Observed Behaviour

Violations:
```verilog
L:   3 | P:  46 | L027 | Unqualified reference 'x' found in select with more than one referenced table/view.
L:   3 | P:  51 | L027 | Unqualified reference 'x' found in select with more than one referenced table/view.
```

### How to reproduce

```sql
select
    t1.id,
    filter(array[t1.col1, t1.col2, t2.col3], x -> x is not null) as flt
from t1
inner join t2 on t1.id = t2.id
```

### Dialect

athena

### Version

1.3.2

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
